### PR TITLE
Implements Abrahamson et al. (2018) "BC Hydro Update" GMPE

### PR DESCRIPTION
**Pull request by g-weatherill**
Addresses https://github.com/gem/oq-engine/issues/3911

An update to the BC Hydro Subduction GMPE is published by Abrahamson et al (2018) (here: https://peer.berkeley.edu/sites/default/files/2018_02_abrahamson_9.10.18.pdf). Although several regional adaptations of the model are described only the "Cascadia-adjusted" version is recommended for usage, and as such only this model is implemented here.

Coefficients are verified against those given in an Excel spreadsheet as a supplement to the document. 

A complete independent implementation for generation of a comprehensive set of test tables has not yet been sourced; however, the Excel spreadsheet itself contains an implementation of the model and test tables have been built from this information by hand. Whilst the number of tests is not as exhaustive as would be ideal, it does cover all of the various source, path and site conditions contained within the GMPE.

### CODE REVIEW

File: openquake/hazardlib/gsim/abrahamson_2018.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    Abrahamson et al. (2018) updated "BC Hydro" subduction GMPE for application
+    to subduction in-slab earthquakes.
+    """
+
+    #: Required rupture parameters for the in-slab model are magnitude and top
+    # of rupture depth
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', 'ztor'))
+
+    #: Cascadia adjustment factor
+    CASCADIA_ADJUSTMENT = "adj_slab"
```
Comment (cvanhoutte) [inline openquake/hazardlib/gsim/abrahamson_2018.py]: Hi there Graeme
Is this missing 
    `DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTRASLAB`
?
Chris
Or similar

Comment (g-weatherill) [reply openquake/hazardlib/gsim/abrahamson_2018.py]: Yes, sorry I overlooked that. Now fixed. Cheers

**Comment by g-weatherill** [comment]
Adding a comment to "bump" this up the priority least: Ongoing investigation of this model is being undertaken with the SERA project and it may still be needed for calculations, ergo there is a time constraint for integration of this feature.

**Comment by micheles** [comment]
LGTM on my side

File: openquake/hazardlib/tests/gsim/abrahamson_2018_test.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    AbrahamsonEtAl2018SInterHigh,
+    AbrahamsonEtAl2018SInterLow,
+    AbrahamsonEtAl2018SSlab,
+    AbrahamsonEtAl2018SSlabHigh,
+    AbrahamsonEtAl2018SSlabLow)
+from openquake.hazardlib.tests.gsim.utils import BaseGSIMTestCase
+
+
+# Interface
+class AbrahamsonEtAl2018SInterTestCase(BaseGSIMTestCase):
```
Comment (mmpagani) [inline openquake/hazardlib/tests/gsim/abrahamson_2018_test.py]: Can you pls add a note specifying how you created the verification tables? Thanks

**Pull request merged**